### PR TITLE
Fix warning about variable macro args with no args

### PR DESCRIPTION
**Pull request by crtrott**
Turns out we have to go through the somewhat more complex
MSVC variant of counting args, in order to avoid the warning.

Addresses #131

### CODE REVIEW

File: tests/test_mdarray_ctors.cpp [hunk 1]
```diff
@@ -117,8 +117,10 @@ struct mdarray_values<2> {
   template<class MDA>
   static void check(const MDA& m) {
     for(int i=0; i<m.extent(0); i++)
-      for(int j=0; j<m.extent(1); j++)
-        ASSERT_EQ(__MDSPAN_OP(m,i,j), 42 + i*1000 + j);
+      for(int j=0; j<m.extent(1); j++) {
+        auto tmp = __MDSPAN_OP(m,i,j);
+        ASSERT_EQ(tmp, 42 + i*1000 + j);
+      }
```
Comment (dalg24) [inline tests/test_mdarray_ctors.cpp:123]: Why did you change this?

Comment (crtrott) [reply tests/test_mdarray_ctors.cpp:123]: nested macro madness.

**Review by dalg24 — APPROVED** [review]
